oauth-keycloak-authorizer/src/main/java/io/strimzi/kafka/oauth/server/authorizer/KeycloakRBACAuthorizer.java

```diff
@@ -611,8 +626,52 @@ private void logDenied(Logger logger, AuthorizableRequestContext context, JsonNo
                 ", resource: " + fromResourcePattern(action.resourcePattern()) + ",\n permissions: " + authz);
     }
 
+
+    /**
+     * Method that performs the POST request to fetch grants for the token.
+     * In case of a connection failure or a non-200 status response this method immediately retries the request if so configured.
+     *
+     * Status 401 does not trigger a retry since it is used to signal an invalid token.
+     * Status 403 does not trigger a retry either since it signals no permissions.
+     *
+     * @param token
+     * @return Grants JSON response
+     */
     private JsonNode fetchAuthorizationGrants(String token) {
 
+        JsonNode response = null;
+        Throwable t = null;
+        int i = 0;
+        while (i <= grantsRetries) {
+            i += 1;
+
+            try {
+                if (t != null) {
+                    log.warn("Failed to fetch grants. Will retry (attempt no. " + i + ")", t);
```

Comment:
Is it really a `warn` level? I mean, the plugin is working exactly as expected (honouring the retry). Given the user has a flaky network (and presumably knows they do, since they configured the retries) is this something we should `warn` about, or just `info`?

Reply:
That's a good point. This code is only executed when fetch retry is enabled so it makes sense that it's INFO.